src/tool_getparam.c

```diff
@@ -2758,13 +2792,38 @@ ParameterError parse_args(struct GlobalConfig *global, int argc,
       bool used;
 
       /* Just add the URL please */
-      result = getparameter("--url", orig_opt, argv[i], &used, global, config);
+      result = getparameter("--url", orig_opt, &toclear, &used, global,
+                            config);
+
+#ifdef HAVE_WIN32_ACMDLN
+      if(tcmdln) {
+        /* !checksrc! disable BANNEDFUNC 2 */
+        _tcscat(tcmdln, TEXT(" "));
+        _tcscat(tcmdln, argv[i]);
+      }
+#endif
     }
 
     if(!result)
       curlx_unicodefree(orig_opt);
   }
 
+#ifdef HAVE_WIN32_ACMDLN
+  if(tcmdln) {
+    size_t acmdln_siz = acmdln_len * sizeof(char);
+    size_t wcmdln_siz = wcmdln_len * sizeof(wchar_t);
+#ifdef UNICODE
+    memcpy(_wcmdln, tcmdln, wcmdln_siz);
+    (void)WideCharToMultiByte(CP_ACP, 0, tcmdln, -1, _acmdln, (int)acmdln_siz,
+                              NULL, NULL);
+#else
+    memcpy(_acmdln, tcmdln, acmdln_siz);
+    (void)MultiByteToWideChar(CP_ACP, 0, tcmdln, -1, _wcmdln, (int)wcmdln_siz);
```

Comment:
I'm pretty sure you're supposed to pass _wcmdln length in wide characters not bytes, refer to [MultiByteToWideChar](https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar), so I think you could do both of these with acmdln_len / wcmdln_len and get rid of acmdln_siz / wcmdln_siz

Reply:
Thanks. I did read the big warning on MSDN for `WildCharToMultiByte()`, and still got it wrong. I hope it's okay now (and also a couple of lines shorter).